tests/utils/v2Data/events.js

```diff
@@ -0,0 +1,310 @@
+        'deleteBucketReplication',
+        'deleteBucketWebsite',
+        'getBucketAcl',
+        'getBucketCors',
+        'getBucketLocation',
+        'getBucketObjectLock',
+        'getBucketReplication',
+        'getBucketVersioning',
+        'getBucketWebsite',
+        'headBucket',
+        'listBucket',
+        'putBucketAcl',
+        'putBucketCors',
+        'PutBucketObjectLock',
+        'putBucketReplication',
+        'putBucketVersioning',
+        'putBucketWebsite',
+    ],
+
+
+    obj: [
+        'deleteObjectTagging',
+        'getObjectAcl',
+        'getObjectLegalHold',
+        'getObjectRetention',
+        'getObjectTagging',
+        'headObject',
+        'putDeleteMarkerObject',
+        'putObjectAcl',
+        'putObjectLegalHold',
+        'putObjectRetention',
+        'putObjectTagging',
+    ],
+
+};
+
+const operationToEventTemplate = Object.entries(eventTemplateToOperation)
+    .reduce((prev, [key, values]) => {
+        values.forEach(v => {
+            prev[v] = key;
+        });
+        return prev;
+    }, {});
+
+const allOperationIds = Object.values(eventTemplateToOperation)
+    .reduce((prev, opIds) => {
+        prev.push(...opIds);
+        return prev;
+    }, []);
+
+function randOperationId(objectExists = true) {
+    if (objectExists) {
+        return randChoice(allOperationIds);
+    }
+    // Weight putting data more heavily so ops that require
+    // the object to exist have a better chance of being used
+    if (Math.random() < 0.75) {
+        return randChoice(
+            eventTemplateToOperation.put,
+        );
+    }
+    return randChoice(
+        eventTemplateToOperation.bucket,
+    );
+}
+
+
+function makeEvent(timestamp, operationId, params) {
+    const template = eventTemplates[[operationToEventTemplate[operationId]]];
+
+    const metric = Object.entries({ ...requiredFields, ...template })
+        .reduce((_metric, [key, factory]) => {
+            _metric[key] = factory(params);
+            // Explicitly remove keys that are undefined
+            if (_metric[key] === undefined) {
+                delete _metric[key];
+            }
+            return _metric;
+        }, {});
+
+    return { timestamp, operationId, ...metric };
+}
+
+
+function getRandAccountUserBucket(objects) {
+    const account = randChoice(Object.keys(objects));
+    const user = randChoice(Object.keys(objects[account]));
+    const bucket = randChoice(Object.keys(objects[account][user]));
+    return { account, user, bucket };
+}
+
+function getRandOrNewKey(existing) {
+    if (existing.length === 0 || Math.random() < 0.1) {
+        return [false, uuid.v4()];
+    }
+    return [true, randChoice(existing)];
+}
+
+const _emptyTotal = {
+    bytes: 0,
+    count: 0,
+    in: 0,
+    out: 0,
+};
+
+function updateTotal(total, event) {
+    if (total === undefined) {
+        total = { ..._emptyTotal };
+        total.ops = {};
+    }
+    total.bytes += event.sizeDelta || 0;
```

Comment:
Wasn't able to find where event is defined here but what is the default if its not 0 here? is it undefined?

Reply:
yes, ops like `putBucketReplication` won't contain a `sizeDelta`